### DISCUSSION TITLE
Insert -> Attachement

### DISCUSSION THREAD

**Discussion by richwein3** [zim-desktop-wiki/zim-desktop-wiki#2524] [Ideas]
I use the insert attachment to bring pdf or other files into my subpage.  It creates a duplicate of the file.  Having a duplicate is not always beneficial, especially if you need to maintain or update the file.   

There is an insert link, which is another way to insert a filesystem item without copy, but leave it where it is.

Is there a 'move' function ?  I didn't see it.

- copy : Insert Attachment exists
- **move**: do not duplicate the file, just move it to the zimwiki subpage directory
- link: Insert Link  exists

Of course I can do a 'move' by doing the copy using insert attachment, then using the OS file browser to delete the original location.

**Comment by fpinkblotch** [comment]
I'd like something similar but from in a drag and drop fashion from the OS file browser. For example:

- Dragging into the text area while holding CTRL copies the file into the attachment directory, and inserts a link at the cursor, pointing to the attachment directory file
- Dragging into the text area while holding SHIFT moves the file into the attachment directory, and inserts the link as well

Wzhat do you think?

**Reply by ZimTom** [reply]
> What do you think?

It all depends on how and in what context you use ZIM, but in general it sounds like a good idea. Your proposal would be a nice addendum to the normal drag'n drop from OS file browser to a ZIM page, which just sets the link while leaving the file where it is.

That said, I would suggest to make this an option you can choose in preferences in order to allow users to adapt to the condition under which ZIM is used. E.g. the option to move entire pages including their attachment directory by drag'n drop can be switched on and off in the configuration of the page index extension and this is really useful. I usually keep it switched off and only switch it on when I need to do some restructuring in my Wiki. This prevents accidental moving of pages and thus unwanted changes of your Wiki structure when working under difficult conditions, e.g. having to work with trackpad while travelling. It also helps to protect your file structure when providing a ZIM-Wiki to people that are a little bit "clumsy" with mouse or trackpad operation. And the same argument applies to your proposal in order to prevent unwanted moving o duplicating attachment files, while the standard drag'n drop function (creating a link to a file) is generally harmless to your Wiki content and the file structure on your volume.
